webhook_notification_test.go

```diff
@@ -249,3 +249,99 @@ func TestWebhookParseDisbursementException(t *testing.T) {
+                    <merchant-account-id>abc123</merchant-account-id>
+                    <reason>fraud</reason>
+                    <reason-code>83</reason-code>
+                    <reason-description>Reason code 83 description</reason-description>
+                    <received-date type="date">2016-02-15</received-date>
+                    <reference-number>123456</reference-number>
+                    <reply-by-date type="date">2016-02-22</reply-by-date>
+                    <status>open</status>
+                    <updated-at type="datetime">2013-04-10T10:50:39Z</updated-at>
+                    <original-dispute-id>original_dispute_id</original-dispute-id>
+                    <status-history type="array">
+                        <status-history>
+                            <status>open</status>
+                            <timestamp type="datetime">2013-04-10T10:50:39Z</timestamp>
+                            <effective-date type="date">2013-04-10</effective-date>
+                        </status-history>
+                    </status-history>
+                    <evidence type="array">
+                        <evidence>
+                            <created-at type="datetime">2013-04-11T10:50:39Z</created-at>
+                            <id>evidence1</id>
+                            <url>url_of_file_evidence</url>
+                        </evidence>
+                        <evidence>
+                            <created-at type="datetime">2013-04-11T10:50:39Z</created-at>
+                            <id>evidence2</id>
+                            <comment>text evidence</comment>
+                            <sent-to-processor-at type="date">2009-04-11</sent-to-processor-at>
+                        </evidence>
+                    </evidence>
+                    <transaction>
+                        <id>123456</id>
+                        <amount>100.00</amount>
+                        <created-at>2017-06-21T20:44:41Z</created-at>
+                        <order-id nil="true"/>
+                        <purchase-order-number nil="true"/>
+                        <payment-instrument-subtype>Visa</payment-instrument-subtype>
+                    </transaction>
+                    <date-opened type="date">2014-03-28</date-opened>
+                    <date-won type="date">2014-04-05</date-won>
+                </dispute>
+            </subject>
+        </notification>`))
+	hmacedPayload, err := hmacer.hmac(payload)
+	if err != nil {
+		t.Fatal(err)
+	}
+	signature := hmacer.publicKey + "|" + hmacedPayload
+
+	notification, err := webhookGateway.Parse(signature, payload)
+
+	if err != nil {
+		t.Fatal(err)
+	} else if notification.Kind != DisputeOpenedWebhook {
+		t.Fatal("Incorrect Notification kind, expected dispute got", notification.Kind)
+	} else if notification.Dispute() == nil {
+		t.Fatal("Notification should have a dispute")
+	} else if notification.Dispute().Kind != DisputeKindChargeback {
```

Comment:
All the assertions past this point don't need to use `Fatalf`. They could use `Errorf` and be standalone `if` statements so that if multiple of them are wrong it will report all of the fields that have failed assertions instead of just the first one.